app/assets/javascripts/app/validate-field.js

```diff
@@ -27,6 +27,7 @@ function addInvalidMarkup(f) {
 
 function removeInvalidMarkup(f) {
   f.parentNode.classList.remove('has-error');
+  f.parentNode.classList.remove('invalid');
```

Comment:
Yeah I agree, there's another open issue that deals with a conflict we're having with the submit button and client-side errors I can link you to.  In this particular case though, with the invalid email address error message coming from the server, no fields actually have the class of `invalid` since the email entered in (user@a.com for example) is a valid email in the mind of the client side validation.  That particular line would also remove it from the parent div and not the actual input field.

![image](https://cloud.githubusercontent.com/assets/1178494/20076570/df040196-a506-11e6-8136-9bfc5df7e2e1.png)

If you want, we could pair up on this one and I can show you what I've found out so far since it's a little hard to explain without seeing and I've spent some time trying to wrangle this one too.